Sources/AsyncHTTPClient/HTTPClient+Proxy.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2018-2019 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHTTP1
+
+extension HTTPClient.Configuration {
+    /// Proxy server configuration
+    /// Specifies the remote address of an HTTP proxy.
+    ///
+    /// Adding an `Proxy` to your client's `HTTPClient.Configuration`
+    /// will cause requests to be passed through the specified proxy using the
+    /// HTTP `CONNECT` method.
+    ///
+    /// If a `TLSConfiguration` is used in conjunction with `HTTPClient.Configuration.Proxy`,
+    /// TLS will be established _after_ successful proxy, between your client
+    /// and the destination server.
+    public struct Proxy {
+        enum ProxyType: Hashable {
+            case http(HTTPClient.Authorization?)
+            case socks
+        }
+
+        /// Specifies Proxy server host.
+        public var host: String
+        /// Specifies Proxy server port.
+        public var port: Int
+        /// Specifies Proxy server authorization.
+        public var authorization: HTTPClient.Authorization? {
```

Comment:
Can we specific specifically that this is HTTP authorization only?

Reply:
This is only moved from a previous file (HTTPClientProxyHandler). Since we don't want to change public API at all, the only adjustments we can make here is in documentation.

Reply:
Yeh this is what I mean - previously there was a comment